### DISCUSSION TITLE
gNB COTS UE's with Raspberry 4/5?

### DISCUSSION THREAD

**Discussion by mianrazo** [srsran/srsRAN_Project#349] [General Help] [answered]
Hi, in the past I was able to complete the system on 4g using srseNB and it was working fine I would like to do the upgrade to 5G now, any of you guys know someone who already try out the new CU/DU architecture for gNB with a raspberry pi 4 or 5 ? if so please let me know your findings , thanks!

**Comment by brendan-mcauliffe** [comment]
Hi @mianrazo this is possible for sure. You should be able to do at least a 5 MHz FDD cell and connect to a COTS UE.

**Reply by mianrazo** [reply]
Hi Brendan, that's good to know, I was considering to try now a Pi 5, do you know of any other small form factor that could work for a 20MHz TDD config ?

**Comment by andrepuschmann** [comment, marked as answer]
The Pi4 will work but it is very limited. I strongly suggest the Pi5. Also see https://github.com/srsran/srsRAN_Project/discussions/339#discussioncomment-7690124 for reference. The Pi5 has the same A76 strong cores but no little cores. So it would be interesting how both compare.

**Reply by mianrazo** [reply]
Hi Andre, thanks for your feedback, I'll try to get it to work on Pi 5 it's nice to know there's already proof this might actually work

**Reply by faratell** [reply]
> Hi Andre, thanks for your feedback, I'll try to get it to work on Pi 5 it's nice to know there's already proof this might actually work

Hi. can you implement 20MHz BW for srsran_4G on Rpi5?How do it? BW=15 work for me but BW=20MHZ show underflow.

**Comment by tilldroemmer** [comment]
Hi @mianrazo did you manage to get it to work?
If so did you only run the gNB on the Raspberry, or did you also try to get the UE to work?
